src/ConnectionDispatcher.hpp

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <memory>
+
+#include "DBConnection2.hpp"
+
+class ConnectionDispatcher {
+public:
+    ConnectionDispatcher() = default;
+    ~ConnectionDispatcher();
+
+    void dispatchConnection(const MySqlDBConnection2& connection) const;
+    const MySqlDBConnection2& getMySqlConnection() const;
+
+private:
+    mutable std::unique_ptr<const MySqlDBConnection2> m_connection;
```

Comment:
@Morheit , I see to much of "const" and mutable. Are you sure you need this?

Reply:
@malirod I doing this way to clarificate that we don't need any changes on this object and we need only use of functions